### PR TITLE
test(e2e): migrate to synpress and playwright

### PR DESCRIPTION
**Pull request by dohaki**
Based on #1075 

Closes ACX-2210

### CODE REVIEW

File: e2e/tests/04_transactions_page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { test, expect } from "@playwright/test";
+import { metaMaskFixtures, testWithSynpress } from "@synthetixio/synpress";
+
+import { E2E_DAPP_URL } from "../config";
+import connectedSetup from "../wallet-setup/connected.setup";
+
+const testWithConnectedMM = testWithSynpress(metaMaskFixtures(connectedSetup));
+const { expect: expectWithConnectedMM } = testWithConnectedMM;
```
Comment (james-a-morris) [inline e2e/tests/04_transactions_page.spec.ts:8]: 'expectWithConnectedMM' is assigned a value but never used

File: src/utils/token.ts [hunk 2]
```diff
@@ -46,9 +48,10 @@ export async function getAllowance(
   owner: string,
   spender: string,
   tokenSymbol: string,
-  blockNumber: number | "latest" = "latest"
+  blockNumber: number | "latest" = "latest",
+  provider?: ethers.providers.Provider
```
Comment (james-a-morris) [inline src/utils/token.ts:52]:
```suggestion
  provider: ethers.providers.Provider = getProvider(chainId)
```

**Review by james-a-morris — COMMENTED** [review]
Did an intial pass. Was the `crypress` directory renamed or moved w/ changes?

**Comment by dohaki** [comment]
> Did an initial pass. Was the `crypress` directory renamed or moved w/ changes?

It was moved into `e2e` folder and migrated to use Playwright. It has better performance and DX and is better supported by Synpress

File: .env.example [hunk 3]
```diff
@@ -63,9 +63,6 @@ REACT_APP_FLAT_RELAY_CAPITAL_FEE=
 # Show rewards banner warning on /rewards page if set to `true`.
 REACT_APP_REWARDS_BANNER_WARNING= 
 
-# Define Hardhat Infura ID for testing
-HARDHAT_INFURA_ID=
```
[the remark this answers is not in the document]
Comment (dohaki) [reply .env.example:67]: Nope, not needed anymore

File: .github/workflows/chromatic.yaml [hunk 4]
```diff
@@ -1,18 +1,14 @@
-# Workflow name
 name: Chromatic
 
 on:
   # manual trigger
   workflow_dispatch:
   pull_request:
 
-# List of jobs
```
[the remark this answers is not in the document]
Comment (dohaki) [reply .github/workflows/chromatic.yaml:9]: Yea true.. But these changes are so small and without any risk that I would just leave them in this PR if you don't mind

File: .github/workflows/e2e.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      github.event_name == 'deployment_status' &&
+      github.event.deployment_status.state == 'success' &&
+      contains(github.event.deployment_status.environment_url, 'app-frontend-v3')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup
+      # For now, we only need Chromium.
+      - name: Install Playwright Browsers
+        run: yarn playwright install chromium
```
[the remark this answers is not in the document]
Comment (dohaki) [reply .github/workflows/e2e.yaml:18]: Synpress still seems very flaky when running in CI with other browsers than Chromium. So I would target one browser for now

File: e2e/config.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  ARBITRUM: 42161,
+  LINEA: 59144,
+  SCROLL: 534352,
+};
+
+export const MM_SEED_PHRASE =
+  process.env.E2E_MM_SEED_PHRASE ||
+  "test test test test test test test test test test test junk";
+export const MM_PASSWORD =
+  process.env.E2E_MM_PASSWORD || "SynpressIsAwesomeNow!!!";
```
[the remark this answers is not in the document]
Comment (dohaki) [reply e2e/config.ts:31]: No, they don't need to be passed necessarily, only if you want to override the defaults. So, no throwing doesn't make sense here

File: e2e/utils/deposit-test-routes.ts [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+    fromTokenAddress: "0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85",
+    toTokenAddress: "0xd988097fb8612cc24eeC14542bC03424c656005f",
+    fromSpokeAddress: "0x6f26Bf09B1C792e3228e5467807a900A503c0281",
+    fromTokenSymbol: "USDC",
+    toTokenSymbol: "USDC.e",
+    isNative: false,
+    l1TokenAddress: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+  },
+];
+export const destinationRoutesToTest = {
```
[the remark this answers is not in the document]
Comment (dohaki) [reply e2e/utils/deposit-test-routes.ts:115]: Yes, I initially tried that but couldn't get it working due to some issues with Synpress/Playwright. This is far from ideal, I agree. But see my comment here https://github.com/across-protocol/frontend-v3/pull/1073/files#diff-8eed85ee069979a8b0de0b9008b1dbde8b2ca940b8344ac554d15e24070d4e5bR8-R19 and this should be a temp workaround

File: e2e/wallet-setup/connected.setup.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  MetaMask,
+  defineWalletSetup,
+  getExtensionId,
+} from "@synthetixio/synpress";
+
+import { MM_PASSWORD, MM_SEED_PHRASE, E2E_DAPP_URL } from "../config";
+
+export default defineWalletSetup(MM_PASSWORD, async (context, walletPage) => {
+  // This is a workaround for the fact that the MetaMask extension ID changes.
+  // This workaround won't be needed in the near future! 😁
```
[the remark this answers is not in the document]
Comment (dohaki) [reply e2e/wallet-setup/connected.setup.ts]: removed https://github.com/across-protocol/frontend-v3/pull/1073/commits/f044391a0f4e2fa4343ae9b8a9da8805e65e9786

**Pull request merged**